pkg/codegen/internal/test/helpers.go

```diff
@@ -102,12 +111,119 @@ func loadDirectory(fs map[string][]byte, root, path string) error {
 	return nil
 }
 
+// Removes files from directory recursively unless the are ignored.
+func removeFilesFromDirUnlessIgnored(root, path string, ignore func(path string) bool) error {
+	entries, err := os.ReadDir(path)
+	if err != nil {
+		return err
+	}
+
+	for _, e := range entries {
+		entryPath := filepath.Join(path, e.Name())
+		relativeEntryPath := entryPath[len(root)+1:]
+
+		if ignore != nil && ignore(relativeEntryPath) {
+			// pass
+		} else if e.IsDir() {
+			if err = removeFilesFromDirUnlessIgnored(root, entryPath, ignore); err != nil {
+				return err
+			}
+		} else {
+			if err = os.Remove(path); err != nil {
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+// Reads `.sdkcodegenignore` file if present to use as loadDirectory ignore func.
```

Comment:
Perhaps trying to be too cute here, something like .gitignore with patterns that exclude files from codegen content-checking and auto-deletion machinery, and especially the PULUMI_ACCEPT=1 machinery.

Files that need this:

- node_modules
- any hand-written test files that are put in alongside the generated files